src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslExportProvider.cs

```diff
@@ -22,40 +22,31 @@ internal OpenSslExportProvider(X509Certificate2Collection certs)
             ICertificatePalCore certificatePal,
             ReadOnlySpan<char> password)
         {
-            AsymmetricAlgorithm? alg = null;
+            AsymmetricAlgorithm alg;
```

Comment:
This can be moved down, and then it becomes:
```C#
AsymmetricAlgorithm alg = evpAlgId switch
{
    Interop.Crypto.EvpAlgorithmId.RSA => new RSAOpenSsl(privateKey),
    Interop.Crypto.EvpAlgorithmId.ECC => new ECDsaOpenSsl(privateKey),
    Interop.Crypto.EvpAlgorithmId.DSA => new DSAOpenSsl(privateKey),
    _ => throw new CryptographicException(SR.Cryptography_InvalidHandle),
};
```